src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -581,7 +587,18 @@ def _run_pytest(self, targets):
       args.extend(test_args)
       args.extend(sources_map.keys())
 
+      # We want to ensure our reporting based off junit xml is from this run so kill results from
+      # prior runs.
+      if os.path.exists(junitxml_path):
+        os.unlink(junitxml_path)
+
       result = self._do_run_tests_with_args(pex, args)
+
+      # There was a problem prior to test execution preventing junit xml file creation so just let
+      # the failure result bubble.
+      if not os.path.exists(junitxml_path):
```

Comment:
Rather than short circuiting here, could the rest of the method body move to the caller? Alternatively, could the rest of the method body move into another method that is just called conditionally?